crates/payload/basic/src/builder_stack.rs

```diff
@@ -0,0 +1,80 @@
+use crate::{
+    BuildArguments, BuildOutcome, MissingPayloadBehaviour, PayloadBuilder, PayloadBuilderError,
+    PayloadConfig,
+};
+
+/// A stack of payload builders that allows for flexible composition of payload builders.
+///
+/// This structure enables the chaining of multiple `PayloadBuilder` implementations,
+/// creating a hierarchical fallback system. It's designed to be nestable, allowing
+/// for complex builder arrangements like `Stack<Stack<A, B>, C>`.
+#[derive(Debug)]
+pub struct PayloadBuilderStack<L, R> {
+    left: L,
+    right: R,
+}
+
+impl<L, R> PayloadBuilderStack<L, R> {
+    /// Creates a new `PayloadBuilderStack` with the given left and right builders.
+    pub const fn new(left: L, right: R) -> Self {
+        Self { left, right }
+    }
+}
+
+impl<L, R, Pool, Client> PayloadBuilder<Pool, Client> for PayloadBuilderStack<L, R>
+where
+    L: PayloadBuilder<Pool, Client>,
+    R: PayloadBuilder<Pool, Client, Attributes = L::Attributes, BuiltPayload = L::BuiltPayload>,
+{
+    type Attributes = L::Attributes;
+    type BuiltPayload = L::BuiltPayload;
```

Comment:
there is now no way to determine which builder should be used, hence we should use `Either<L::Attr, R::Atrr>` here

ref https://github.com/paradigmxyz/reth/pull/11004/files#r1768054937